odoo/addons/base/models/ir_fields.py

```diff
@@ -344,7 +344,11 @@ def db_id_for(self, model, field, subfield, value):
             else:
                 xmlid = "%s.%s" % (self._context.get('_import_current_module', ''), value)
             flush(xml_id=xmlid)
-            id = self.env['ir.model.data'].xmlid_to_res_id(xmlid, raise_if_not_found=False) or None
+            # _lookup_xmlids() finds the external id and the record id if
+            # present in the model's table, and returns a list of tuples
+            # (imd_id, module, name, model, res_id, noupdate, record_id)
+            imd_data = self.env['ir.model.data']._lookup_xmlids([xmlid], RelatedModel)
+            id = imd_data[0][-1] if imd_data else None
```

Comment:
Maybe the comment should be more explicit that the `record_id` here is joined-in and will thus be `None` if the xid is dangling (otherwise it should be the same as `res_id`)?